lib/membrane_rtc_engine/endpoints/webrtc/track_sender.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Membrane.RTC.Engine.Endpoint.WebRTC.TrackSender do
+  @moduledoc false
+
+  # TrackSender:
+  # * adds `is_keyframe` flag to each buffer's metadata
+  # (will be removed after releasing new RTP plugin)
+  # * tracks encoding activity
+
+  use Membrane.Filter
+
+  require Membrane.Logger
+
+  alias Membrane.Buffer
+  alias Membrane.RTC.Engine.Track
+
+  def_options track: [
+                type: :struct,
+                spec: Membrane.RTC.Engine.Track.t(),
+                description: "Track this sender will maintain"
+              ]
+
+  def_input_pad :input,
+    availability: :on_request,
+    mode: :pull,
+    demand_mode: :auto,
+    caps: Membrane.RTP
+
+  def_output_pad :output,
+    availability: :on_request,
+    mode: :pull,
+    demand_mode: :auto,
+    caps: Membrane.RTP
+
+  @impl true
+  def handle_init(%__MODULE__{track: track}) do
+    {:ok, %{track: track}}
+  end
+
+  @impl true
+  def handle_pad_added(Pad.ref(:input, {_track_id, _rid}), _ctx, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_pad_added(
+        Pad.ref(:output, {_track_id, _rid}) = pad,
+        %{playback_state: :playing},
+        state
+      ) do
+    {{:ok, caps: {pad, %Membrane.RTP{}}}, state}
+  end
+
+  @impl true
+  def handle_pad_added(_pad, _ctx, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_caps(_pad, _caps, _ctx, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_prepared_to_playing(ctx, state) do
+    actions =
+      Enum.flat_map(ctx.pads, fn
+        {Pad.ref(:output, _ref) = pad, _pad_data} -> [caps: {pad, %Membrane.RTP{}}]
+        _other -> []
+      end)
+
+    {{:ok, actions}, state}
+  end
```

Comment:
IIRC the forward action will do the same i.e. sends it on all the output pads, That is also a default behaviour of `handle_caps`, so you can take advantage of that

Reply:
Looks like forward action cannot be used in `handle_prepared_to_playing`. I cannot also use `handle_caps` as I would duplicate caps on output pads.

Reply:
You're right, however, it is possible to use forward in `handle_caps` by getting rid of this (`handle_prepared_to_playing`) callback and changing the condition in `handle_pad_added` - instead of checking the playback state, send caps only when they are set in a context for the input pad - then `handle_caps` would send the caps to any pad added before receiving caps and `handle_pad_added` from that moment afterwards. Plus `pad_added` could use the received caps instead of generating them. However, I'm not sure if that's more readable, I'll leave it up to you to decide

Reply:
I would keep current version for now as we have the same implementation in SimulcastTee